ssl/t1_lib.c

```diff
@@ -2642,6 +2642,43 @@ static int sigalg_security_bits(SSL_CTX *ctx, const SIGALG_LOOKUP *lu)
     return secbits;
 }
 
+static int tls_sigalg_compat(SSL_CONNECTION *sc, const SIGALG_LOOKUP *lu)
+{
+    int minversion, maxversion;
+    int minproto, maxproto;
+
+    if (!lu->available)
+        return 0;
+
+    if (SSL_CONNECTION_IS_DTLS(sc)) {
+        if (sc->ssl.method->version == DTLS_ANY_VERSION) {
+            minproto = sc->min_proto_version;
+            maxproto = sc->max_proto_version;
+        } else {
+            maxproto = minproto = sc->version;
+        }
```

Comment:
It seems this `if` could be moved outside of the DTLS conditional. It's virtually identical to the TLS variant except we check `DTLS_ANY_VERSION` instead of `TLS_ANY_VERSION`

Reply:
I may do that later, for now let's settle the non-cosmetic issues.

Reply:
I've decided to not do that. You may not have noticed that the `minversion` and `maxversion` assignments from `lu` are also dependent on the DTLS vs. TLS protocol status.